src/librustc_typeck/check/method/suggest.rs

```diff
@@ -69,13 +69,27 @@ pub fn report_error<'a, 'tcx>(fcx: &FnCtxt<'a, 'tcx>,
             }
 
             if !static_sources.is_empty() {
-                fcx.tcx().sess.fileline_note(
+                cx.sess.fileline_note(
                     span,
                     "found defined static methods, maybe a `self` is missing?");
 
                 report_candidates(fcx, span, item_name, static_sources);
             }
 
+            if !predicates.is_empty() {
+                let bound_list = predicates.iter()
+                    .map(|p| format!("`{} : {}`",
+                                     p.substs.self_ty().unwrap(),
+                                     p))
+                    .collect::<Vec<_>>().connect(", ");
+                cx.sess.fileline_note(
+                    span,
+                    &format!("The method `{}` exists but the following trait bounds were not \
```

Comment:
Nit: error messages and the like begin with lower-case letters

Nit: this line is pretty long; given that you're wrapping the string constant anyway, can you put the newline break earlier? I'd put it after "but the" -- mainly because I try to group "important words" like "following trait bounds" on the same line so that later when I ack for an error message I can find it more easily.